src/client/Microsoft.Identity.Client/PlatformsCommon/Shared/DesktopOsHelper.cs

```diff
@@ -82,35 +80,53 @@ public static bool IsMac()
 #endif
         }
 
-        public static bool IsWin10()
+        /// <summary>
+        /// Checks the retrieved Windows version to see if it is a WAM supported Windows Client OS. 
+        /// We get the OS Version from the registry and Windows 11 Product name is listed as Windows 10
+        /// So we are just checking for the "Windows 10" version string. This is by design 
+        /// https://microsoft.visualstudio.com/OS/_workitems/edit/36426651
+        /// </summary>
+        /// <returns>Returns <c>true</c> if the Windows Client Version has WAM support</returns>
+        private static bool IsWin10OrNewerInternal()
         {
-            return s_win10Lazy.Value;
+            string winVersion = GetWindowsVersionString();
+
+            if (winVersion.Contains("Windows 10", StringComparison.OrdinalIgnoreCase))
```

Comment:
Is the value same across all cultures and languages?

Reply:
Tested one locale and it is in English.